app/uk/gov/hmrc/agentmappingfrontend/audit/AuditService.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentmappingfrontend.audit
+
+import javax.inject.{Inject, Singleton}
+
+import play.api.mvc.Request
+import uk.gov.hmrc.agentmappingfrontend.audit.AgentFrontendMappingEvent.AgentFrontendMappingEvent
+import uk.gov.hmrc.domain.SaAgentReference
+import uk.gov.hmrc.play.audit.AuditExtensions.auditHeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.audit.model.DataEvent
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.Try
+
+object AuditService {
+
+  def auditCheckAgentRefCodeEvent(saAgentReferenceOpt: Option[SaAgentReference], authProviderId: Option[String], authProviderType: Option[String])
+                                 (auditService: AuditService)
+                                 (implicit hc: HeaderCarrier, request: Request[Any]): Unit = {
+    val event = createEvent(AgentFrontendMappingEvent.CheckAgentRefCode,
+      Seq("isEnrolledSAAgent" -> saAgentReferenceOpt.isDefined)
+        ++ saAgentReferenceOpt.map(r => Seq("saAgentRef" -> r.value)).getOrElse(Seq.empty)
+        ++ authProviderId.map(v => Seq("authProviderId" -> v)).getOrElse(Seq.empty)
+        ++ authProviderType.map(v => Seq("authProviderType" -> v)).getOrElse(Seq.empty)
+    )
+    auditService.send(event)
+  }
+
+  private def createEvent(event: AgentFrontendMappingEvent,
+                          details: Seq[(String, Any)])
+                         (implicit hc: HeaderCarrier, request: Request[Any]): DataEvent = {
+    DataEvent(
+      auditSource = "agent-mapping-frontend",
+      auditType = event.toString,
+      tags = hc.toAuditTags("", request.path),
```

Comment:
Ok, according to [Auditing Best Practice Guide](https://confluence.tools.tax.service.gov.uk/display/DTRG/Auditing+Best+Practice+Guide)

This field is intended as a human readable identifier for the operation being performed. Defaults to the tags.path, but should be set explicitly. If the particular operation/user action being processed changes significantly (re-factored or aggregated for example) the transactionName should probably change too. Think about the semantics of the operation/action and what data the audit event encapsulates. If these change then the transacationName should as well.
Transaction name should be set explicitly and have a meaningful name in the context of the user journey and operation.  Relying on the implicit value (which is the URI stem of the micro service called) is fragile and not recommended.  Micro-service URIs can change with time.
Example : poison_password_provisioned